sherpa without sherpa.

r2ishiguro opened the pull request.

Commit:
sherpa without sherpa.

maditya commented on httpfront/httpfront.go:
```diff
@@ -103,6 +104,7 @@ func (h *HTTPFront) doLookup(b io.Reader, ctx context.Context) (*proto.LookupPro
 	if err != nil {
 		return nil, err
 	}
+fmt.Printf("lookup: %v\n", lr);
```
can we remove this?

maditya commented on lookup.go:
```diff
@@ -75,6 +77,9 @@ func VerifyLookup(cfg *proto.Config, user string, pf *proto.LookupProof, now tim
 		return nil, nil
 	} else {
+		if pf.Entry == nil || pf.Profile == nil {
+			return nil, nil		// empty entry or profile
```
shouldn't this return an error?

maditya commented on lookup.go:
```diff
@@ -266,3 +271,29 @@ func (n *ReconstructedNode) Index() []byte {
 	return n.value
 }
+
+// getUpdate returns the last update to profile of idx during or before epoch.
```
s/getUpdate/GetUpdate?

maditya commented on proto/AuthorizationPolicypr_test.go:
```diff
@@ -15,8 +15,8 @@
 import (
-	github_com_gogo_protobuf_jsonpb "github.com/maditya/protobuf/jsonpb"
 	github_com_gogo_protobuf_proto "github.com/maditya/protobuf/proto"
+	github_com_gogo_protobuf_jsonpb "github.com/maditya/protobuf/jsonpb"
```
shouldn't this point to gogo/protobuf ?

r2ishiguro replied:
Can I remove generated files? It's confusing.

Commit:
remove a debugging message.

Commit:
check Entry and Profile in the lookup proof to avoid the crash.

Commit:
make Marshal, Unmarshal work for made up Encoded*.
skip tests that invole GoString() and String() because non deterministic iteration of map.

Commit:
match the parameters.

Commit:
move ratification policy to server.go following the TODO comment.

maditya commented on keyserver/server_test.go:
```diff
@@ -1203,18 +1205,18 @@ func TestKeyserverHTTPFrontLookup(t *testing.T) {
 	// as specified in https://github.com/yahoo/coname/blob/master/proto/client.proto#L63-L89
 	type lp struct {
-		Entry      string `json:"entry"`
+		Entry      json.RawMessage `json:"entry"`
```
why did we have to change this?

r2ishiguro replied:
because it's not a struct, not a string.

maditya commented on proto/generate.sh:
```diff
@@ -49,3 +50,5 @@ awk '{
     print($0)
 }' < client.pb.go > client.pb.go.tmp && mv client.pb.go.tmp client.pb.go
+
+for i in *.pb.go; do sed "s/\(NewPopulatedEncoded[A-Za-z_0-1]*\)(\(.*\))/\1(\2, easy)/" < $i > $i...; mv -f $i... $i; done
```
can you add a comment on what this is trying to do?

maditya commented on proto/replication.proto:
```diff
@@ -53,10 +53,18 @@ message KeyserverStep {
+
+message EpochUpdate {
+	EpochDelimiter delimiter = 1;
+	bool update = 2;
```
can you add a comment on the purpose of this field?

maditya commented on proto/replication.proto:
```diff
@@ -53,10 +53,18 @@ message KeyserverStep {
+message EpochUpdate {
+	EpochDelimiter delimiter = 1;
+	bool update = 2;
+	fixed64 UID = 3;
```
can you add a comment on the purpose of this field?

Commit:
remove sherpa depending parts.
skip the timestamp test.

Commit:
fix the vrf flow; fix the commitment error

Commit:
use the new vrf function.
fix the commitment bug.

Commit:
implement a new algorithm.

Commit:
change the path.

Commit:
update the test vectors following the latest algorithm.

Commit:
avoid dead lock

Commit:
applied the tlsconfig change.

Commit:
Merge branch 'sherpa' of https://github.com/r2ishiguro/coname into r2ishiguro-sherpa

Commit:
Merge branch 'r2ishiguro-sherpa'

Commit:
Merge branch 'r2ishiguro-sherpa' into sherpa

Commit:
Merge branch 'master' into sherpa

Commit:
Merge remote-tracking branch 'upstream/master' into sherpa
